libraries/python/classification_compare.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+##############################################################################
+# Copyright 2017-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+##############################################################################
+
+# this library is to compare the output of the benchmark and the golden output
+# for image classification tasks, if the golden is 1, expecting the benchmark
+# is the closest to that.
+
+
+import argparse
+import json
+import os
+
+
+parser = argparse.ArgumentParser(description="Output compare")
+
+parser.add_argument("--benchmark-output", required=True,
+    help="The output of the benchmark.")
+parser.add_argument("--labels", required=True,
+    help="The golden output.")
+parser.add_argument("--metric-keyword",
+    help="The keyword prefix each metric so that the harness can parse.")
+parser.add_argument("--result-file",
+    help="Write the prediction result to a file for debugging purpose.")
+parser.add_argument("--top", type=int, default=1,
+    help="Integer indicating whether it is a top one or top five.")
+
+
+class OutputCompare(object):
+    def __init__(self):
+        self.args = parser.parse_args()
+        assert os.path.isfile(self.args.benchmark_output), \
+            "Benchmark output file {} doesn't exist".format(
+                self.args.benchmark_output)
+        assert os.path.isfile(self.args.labels), \
+            "Labels file {} doesn't exist".format(
+                self.args.labels)
+
+    def getData(self, filename):
+        with open(filename, "r") as f:
+            content = f.read()
+            batches = content.strip().split('\n')
+        # separate out for debugging purpose
+        '''
+        data = []
+        for batch in batches:
+            one_batch = batch.strip().split(',')
+            print(batch)
+            import pdb; pdb.set_trace()
+            one_data = []
+            for item in one_batch:
+                print(item + " <--")
+                one_data.append(float(item.strip()))
+            data.append(one_data)
+        '''
+        data = [[float(item) for item in batch.strip().split(',')]
+                for batch in batches]
+        return data
+
+    def writeOneResult(self, values, data, metric, unit):
+        entry = {
+            "type": "model",
+            "values": values,
+            "summary": {
+                "num_runs": len(values),
+                "p0": data,
```

Comment:
This sounds confusing. why do you give statistics on data?

Reply:
We only have one data point. So everything is the same. If we don't, those fields will be empty, which one is better?